core/trino-parser/src/main/java/io/trino/sql/jsonpath/tree/PredicateCurrentItemVariable.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.jsonpath.tree;
+
+public class PredicateCurrentItemVariable
```

Comment:
What does this represent?

Reply:
This is a variable used by filter to access the currently processed item. It is represented by `@` in the path.

Like every accessor, filter takes a sequence of items, and processes them in order. `@` is a reference to the currently processed item. E.g. 
`<json_array>[*] ? (@ > 0)`